pkg/minishift/openshift/service.go

```diff
@@ -136,33 +180,92 @@ func getValidNamespaces(serviceListNamespace string) ([]string, error) {
 	return namespaces, nil
 }
 
-func getServiceURLsOutput(namespace string) (string, error) {
-	cmdArgText := fmt.Sprintf("get route -n %s --config=%s %s", namespace, systemKubeConfigPath, URLsCustomCol)
+func getRouteSpec(namespace string) (map[string][]ServiceWeight, error) {
+	routeSpec := make(map[string][]ServiceWeight)
+	cmdArgText := fmt.Sprintf("get route -o json --config=%s -n %s", systemKubeConfigPath, namespace)
 	tokens := strings.Split(cmdArgText, " ")
 	cmdName := instanceState.InstanceConfig.OcPath
 	cmdOut, err := runner.Output(cmdName, tokens...)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
-	return byteArrayToString(cmdOut), nil
+	if strings.Contains(string(cmdOut), "No resources found") {
+		return nil, nil
+	}
+
+	var data Route
+	err = json.Unmarshal(cmdOut, &data)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, item := range data.Items {
+		totalWeight := item.Spec.To.Weight
+		for _, alternateBackend := range item.Spec.AlternateBackends {
+			totalWeight += alternateBackend.Weight
+		}
+		routeSpec[item.Spec.Host] = []ServiceWeight{{Name: item.Spec.To.Name, Weight: calculateWeight(float64(item.Spec.To.Weight), float64(totalWeight))}}
+		for _, alternateBackend := range item.Spec.AlternateBackends {
+			routeSpec[item.Spec.Host] = append(routeSpec[item.Spec.Host], ServiceWeight{Name: alternateBackend.Name,
+				Weight: calculateWeight(float64(alternateBackend.Weight), float64(totalWeight))})
+		}
+	}
+	return routeSpec, nil
 }
 
-func filterAndUpdateServiceURLS(data, namespace string) []ServiceURL {
-	var serviceURLs []ServiceURL
+func calculateWeight(a float64, b float64) string {
+	weightPercentage := int(a / b * 100)
+	if weightPercentage != 100 {
+		return strconv.Itoa(weightPercentage) + "%"
+	}
+	return ""
+}
 
-	re_whtsp_inside := regexp.MustCompile(`[\s\p{Zs}]{2,}`)
-	data = re_whtsp_inside.ReplaceAllString(data, " ") // replace all extra whitespaces
-	contents := strings.Split(data, "\n")              // split on new lines
-	contents = emptyFilter(contents[1:])               // remove the header "NAME HOST" and empty elements
+func filterAndUpdateServiceSpecs(routeSpec map[string][]ServiceWeight, namespace string, serviceNodePort map[string]string) []ServiceSpec {
+	var serviceSpecs []ServiceSpec
+	serviceSpecMap := make(map[string]*ServiceSpec)
+
+	for routeURL, v := range routeSpec {
+		for _, service := range v {
+			// split content on colon to separate NAME, HOST and Weight
```

Comment:
@hferentschik I did try to have it in a separate method but all those are too dependent that I endup having those in single place,